scipy/special/_basic.py

```diff
@@ -1774,7 +1777,17 @@ def lpmn(m, n, z):
                 fixarr = where(mf > nf, 0.0, gamma(nf-mf+1) / gamma(nf+mf+1))
     else:
         mp = m
-    p, pd = _specfun.lpmn(mp, n, z)
+
+    z = np.asarray(z)
+
+    p = np.empty((mp + 1, n + 1) + z.shape, dtype=np.float64)
+    pd = np.empty_like(p)
+    if isscalar(z):
```

Comment:
If `z` was passed as a scalar then `z = np.asarray(z)` will convert it to an array and `isscalar(z)` will evaluate to `False`. You should check for `z.ndim > 0` to see if it's an array.

Reply:
> If `z` was passed as a scalar then `z = np.asarray(z)` will convert it to an array and `isscalar(z)` will evaluate to `False`. You should check for `z.ndim > 0` to see if it's a non zero d array.

Reply:
Duh, of course. Done.